### PR TITLE
Quic transport is no longer experimental in .NET 6

### PR DESCRIPTION
**Pull request by JamesNK**
**Quic transport is no longer experimental in .NET 6**

This PR is a YOLO find and replace to remove `Experimental` from the Quic transport project and namespaces. I'd apprietate guidance on any ritutuals that are required when removing and adding packages from aspnetcore.

Two things:
1. The runtime package we depend on still has Experimental in the name: `System.Net.Experimental.MsQuic`.
2. I think we should consider updating the project name and namespace to `Microsoft.AspNetCore.Server.Kestrel.Transport.MsQuic` (note: `Quic` -> `MsQuic`). That's consistent with runtime's name. And in the future the Quic implementation might change to be written in managed code so we should be more specific.

PR
* Renames and removes `Experimental` from Quic transport project
* Removes `Experimental` from multiplex connection abstractions types

APIs promoted from experimental status need API review.

### CODE REVIEW

File: src/Servers/Kestrel/Transport.Quic/src/PublicAPI.Shipped.txt [hunk 1]
```diff
@@ -1,24 +1,24 @@
 #nullable enable
 Microsoft.AspNetCore.Hosting.WebHostBuilderMsQuicExtensions
-Microsoft.AspNetCore.Server.Kestrel.Transport.Experimental.Quic.QuicTransportOptions
```
Comment (dougbu) [inline src/Servers/Kestrel/Transport.Quic/src/PublicAPI.Shipped.txt:3]: Do not change this file. Instead copy the lines you deleted into PublicAPI.Unshipped.txt with a `*Removed*` prefix

Comment (dougbu) [reply src/Servers/Kestrel/Transport.Quic/src/PublicAPI.Shipped.txt:3]: This change is why the CodeCheck job failed

**Review by jkotalik — APPROVED** [review]

File: src/Servers/Connections.Abstractions/src/PublicAPI.Unshipped.txt [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-Microsoft.AspNetCore.Connections.Experimental.MultiplexedConnectionContext
-Microsoft.AspNetCore.Connections.Experimental.MultiplexedConnectionContext.MultiplexedConnectionContext() -> void
-Microsoft.AspNetCore.Connections.Experimental.MultiplexedConnectionDelegate
-abstract Microsoft.AspNetCore.Connections.Experimental.MultiplexedConnectionContext.AcceptAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) -> System.Threading.Tasks.ValueTask<Microsoft.AspNetCore.Connections.ConnectionContext?>
-abstract Microsoft.AspNetCore.Connections.Experimental.MultiplexedConnectionContext.ConnectAsync(Microsoft.AspNetCore.Http.Features.IFeatureCollection? features = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) -> System.Threading.Tasks.ValueTask<Microsoft.AspNetCore.Connections.ConnectionContext!>
+Microsoft.AspNetCore.Connections.IMultiplexedConnectionBuilder
```
Comment (jkotalik) [inline src/Servers/Connections.Abstractions/src/PublicAPI.Unshipped.txt:4]: We need to do a formal API review on these.

**Comment by JamesNK** [comment]
**Review notes:**

IMultiplexedConnectionListenerFactory:
- BindAsync IFeatureCollection used?
- Indented for per endpoint cert?

IMultiplexedConnectionListener:
- AcceptAsync IFeatureCollection used?

MultiplexedConnectionContext:
- `ValueTask<ConnectionContext> ConnectAsync`, change return to nullable
- Debate about ConnectAsync name. Options:
   - OpenAsync
   - CreateAsync

QuicTransportFactory:
- Make internal. Likely was made public to match sockets transport

WebHostBuilderMsQuicExtensions:
- Rename to WebHostBuilderQuicExtensions

QuicTransportOptions:
- Revist everything

Will remove experimental now and make changes through previews. Re-review later in .NET 6.

@jkotalik @Tratcher @halter73 @davidfowl @wtgodbe

**Comment by halter73** [comment]
The IFeatureCollection arguments are intended for future extensibility. Since TLS becomes a transport concern with QUIC using it for certs sounds like a viable option.

Why does ConnectAsync need to return a nullable?

**Comment by JamesNK** [comment]
> Why does ConnectAsync need to return a nullable?

@jkotalik added that feedback. Is it to make ConnectAsync return consistent with AcceptAsync?

**Comment by JamesNK** [comment]
@dougbu The CI build is failing with this message:

> eng\Baseline.Designer.props(,): error : Generated code is not up to date in eng/Baseline.Designer.props. You might need to regenerate the reference assemblies or project list (see docs/ReferenceAssemblies.md and docs/ReferenceResolution.md)

What changes do I need to make to pass CI build?

**Comment by dougbu** [comment]
> What changes do I need to make to pass CI build?

It's mostly not you @JamesNK (and this likely applies to every PR that targets 'main' and not-yet-verified):
1. I need to get #30781 in
2. Bot needs to create a merge PR into 'main'
3. I need to get _that_ PR in
4. You need to rebase on latest 'main'

**Comment by jkotalik** [comment]
> @jkotalik added that feedback. Is it to make ConnectAsync return consistent with AcceptAsync?

Yeah that was the idea behind it. Though maybe throwing for either if they fail to connect or accept makes more sense?

**Comment by JamesNK** [comment]
@dougbu @JunTaoLuo @wtgodbe Good to merge from a build POV?

**Review by wtgodbe — APPROVED** [review]
LGTM build-wise

**Review by halter73 — APPROVED** [review]

**Comment by dougbu** [comment]
Side question: Why do the Baseline files appear in the final commit❔ They should have been updated as part of a merge from release/5.0.

**Pull request merged**